### PR TITLE
Addressed issues 192-194

### PR DESCRIPTION
**Pull request by rachlobay**
Since Issues https://github.com/cmu-delphi/epiprocess/issues/192, https://github.com/cmu-delphi/epiprocess/issues/193, and https://github.com/cmu-delphi/epiprocess/issues/194 are closely related, addressed all three here.

After discussion with Logan, made the following changes to `[.epi_df` code for the written issue:
- Issue https://github.com/cmu-delphi/epiprocess/issues/192: Any key columns that have been selected out from an `epi_df` are removed from other_keys in the metadata (refer to the ex in that issue).
- Issue https://github.com/cmu-delphi/epiprocess/issues/193: Dropped all metadata when decay to tibble.
- Issue https://github.com/cmu-delphi/epiprocess/issues/194: Abort if column duplication in an `epi_df` (to be consistent with tibble behaviour).

### CODE REVIEW

File: R/methods-epi_df.R [hunk 1]
```diff
@@ -80,13 +80,39 @@ summary.epi_df = function(object, ...) {
   }
```
Comment (brookslogan) [inline R/methods-epi_df.R:80]: Looks like `i_arg` and `j_arg` aren't used anymore; we can remove them entirely

File: R/methods-epi_df.R [hunk 2]
```diff
@@ -80,13 +80,39 @@ summary.epi_df = function(object, ...) {
...
+  # Duplicate columns, Abort
+  dup_col_names = cn[duplicated(cn)]
+  if (length(dup_col_names) != 0) {
+    Abort(paste0("Column name(s) ", 
+                 paste(unique(dup_col_names), 
+                       collapse = ", "), " must not be duplicated."))
+  } 
+  
+  # If not an epi_df (missing time_value or geo_value), return a tibble
+  not_epi_df <- (!("time_value" %in% cn) || !("geo_value" %in% cn))
```
Comment (brookslogan) [inline R/methods-epi_df.R:93]: Style thing: remove outer parens?  (Inner parens may also be unnecessary, but still might help clarify.)

File: R/methods-epi_df.R [hunk 3]
```diff
@@ -80,13 +80,39 @@ summary.epi_df = function(object, ...) {
...
   
-  # Use reclass as safeguard (in case class & metadata are dropped)
+  if (not_epi_df) {
+    attributes(res)$metadata <- NULL 
+    return(tibble::as_tibble(res))
+  }
+  
+  # Use reclass as safeguard (in case class &/or metadata are dropped)
   reclass(res, attr(x, "metadata"))
```
Comment (brookslogan) [inline R/methods-epi_df.R:101]: I don't think this will actually alter `res`; you probably need `res <- reclass(res, attr(x, "metadata"))`

File: R/methods-epi_df.R [hunk 4]
```diff
@@ -80,13 +80,39 @@ summary.epi_df = function(object, ...) {
...
+  
+  # Extract the other_keys colnames in the original epi_df that do not occur in the subset
+  only_in_orig = setdiff(unlist(attr(x, "metadata")$other_keys), 
+                         cn[!(cn %in% c("geo_value", "time_value"))]) 
+  
+  if (length(only_in_orig) != 0) {
+    # remove all other_keys elements in the original epi_df that do not occur in subset
+    attr(res, "metadata")$other_keys = 
+      attr(res, "metadata")$other_keys[! attr(res, "metadata")$other_keys %in% only_in_orig]
+  }
```
Comment (brookslogan) [inline R/methods-epi_df.R:113]: I think (since we're not messaging about this inside the `if` statement) this section might be able to be written shorter as
```
attr(res, "metadata")$other_keys <- intersect(attr(res, "metadata")$other_keys, cn)
```
and I believe this preserves the original order of the keys that stick around, but `?intersect` doesn't seem to guarantee this.  So if we want to guarantee this behavior, maybe
```
attr(res, "metadata")$other_keys <- attr(res, "metadata")$other_keys[attr(res, "metadata")$other_keys %in% cn]
```
plus a comment about why we're not using `intersect`?

File: R/methods-epi_df.R (hunk 2 above)
Comment (brookslogan) [reply R/methods-epi_df.R:93]: Could you please add some test_that tests for the behaviors above?  I assume at least the duplicate-col error must be missing from the tests.

**Review by brookslogan — CHANGES_REQUESTED** [review]
Couple a minor things + a request to add any relevant tests for the new checks.  Are all the rename-related things already split out into other issues?

**Comment by brookslogan** [comment]
Also, misc thing: it looks like git might be using attaching your computer username to the commits here, which might be a bit confusing looking at git logs.  Could you please try running either:
(1)
```sh
git config user.name "<Your Name>"
git config user.email "<email-addr>@<.....>.edu"
```
in each cloned repo, or
(2)
```sh
git config --global user.name "<Your Name>"
git config --global user.email "<your-email-address>@<.....>.edu"
```
from anywhere?

You can check for success using
```
git config --list
```
in each repo, or
```
git config --global --list
```
to view just the global settings.

File: R/methods-epi_df.R (hunk 2 above)
Comment (rachlobay) [reply R/methods-epi_df.R:93]: Added tests for duplicate-col error and missing `geo_value` col.

**Comment by rachlobay** [comment]
Since the rename issues that were mentioned in https://github.com/cmu-delphi/epiprocess/issues/192 and https://github.com/cmu-delphi/epiprocess/issues/194 were related, I put them into one issue [here](https://github.com/cmu-delphi/epiprocess/issues/195), where I added a little more info based on our last discussion.

File: R/methods-epi_df.R (hunk 4 above)
Comment (rachlobay) [reply R/methods-epi_df.R:113]: After looking into `intersect`, it looks like the order of the resulting elements is the same as the order of the elements in the 1st argument. So, we can use that. Also, `cn = names(res)`... so I think what you meant for the 1st argument in `intersect` is is `attr(x, "metadata")$other_keys` (b/c one of the args should be that)?

**Review by dajmcdon — APPROVED** [review]
LGTM once all @brookslogan comments are addressed.

File: R/methods-epi_df.R (hunk 4 above)
Comment (brookslogan) [reply R/methods-epi_df.R:104]: Right, `intersect` in my version of R also appears to keep the first arg's order; however, I can't find anywhere in the docs that guarantees that this will be the case for any version of R.  That's why I was still looking at that `%in%` solution; in case for some reason they decide to change `intersect`'s implementation within the bounds of the docs.

I think I wrote it assuming that `attr(res, "metadata")` was set to `attr(x, "metadata")`; if that's not the case, then yes, sounds like something should be changed to `x` in there.

Comment (brookslogan) [reply R/methods-epi_df.R:104]: Pushing an update to address this wild fear + to simplify a little bit (we don't even need the if statement!)

**Review by brookslogan — APPROVED** [review]
Addressed a minor check comment.  If checks pass, think this is ready to merge.
